boring/src/x509/verify.rs

```diff
@@ -111,6 +111,34 @@ impl X509VerifyParamRef {
         }
     }
 
+    /// Sets the time to check certificates and CRLs against.
+    ///
+    /// If unset, uses the current time.
+    ///
+    /// This corresponds to [`X509_VERIFY_PARAM_set_time`]. Note that BoringSSL does not support
+    /// the OpenSSL `NO_CHECK_TIME` option.
+    ///
+    /// [`X509_VERIFY_PARAM_set_time`]: https://www.openssl.org/docs/man1.1.1/man3/X509_VERIFY_PARAM_set_time.html
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use boring::x509::store::X509StoreBuilder;
+    /// use std::convert::TryInto;
+    /// use std::time::SystemTime;
+    ///
+    /// let mut store_builder = X509StoreBuilder::new().expect("can create a new builder");
+    /// let duration_since_epoch = SystemTime::UNIX_EPOCH.elapsed().expect("it's after 1970");
+    /// let seconds_since_epoch: libc::time_t = duration_since_epoch
+    ///     .as_secs()
+    ///     .try_into()
+    ///     .expect("time_t is large enough to represent this time");
+    /// store_builder.param_mut().set_time(seconds_since_epoch);
+    /// ```
+    pub fn set_time(&mut self, unix_time: time_t) {
```

Comment:
I don't love directly exposing `time_t` here, but the alternative would be to take a SystemTime directly and fail on converting to `time_t`, which didn't seem consistent with other APIs in this crate.